### PR TITLE
Check the sha when installing opa

### PR DESCRIPTION
**Pull request by simonbaird**
As suggested by @lcarva in #1.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,11 +24,17 @@ fmt-check:
 ci: fmt-check test
 
 OPA_VER=v0.39.0
-OPA_URL=https://openpolicyagent.org/downloads/$(OPA_VER)/opa_linux_amd64_static
+OPA_FILE=opa_linux_amd64_static
+OPA_URL=https://openpolicyagent.org/downloads/$(OPA_VER)/$(OPA_FILE)
+OPA_SHA_FILE=$(OPA_FILE).sha256
+OPA_SHA_URL=$(OPA_URL).sha256
```
Comment (lcarva) [inline Makefile:30]: I was thinking more of including the checkusm in git so we protect ourselves against `openpolicyagent.org` being compromised. The current version wouldn't detect if someone replaced the opa binary with some malicious binary and also replace the digest. It would help in detecting a corrupt file due to download issues, though.

Comment (simonbaird) [reply Makefile:30]: Ah yeah ok. Let me try again.

File: Makefile [hunk 2]
```diff
@@ -24,11 +24,17 @@ fmt-check:
...
 OPA_DEST=/usr/bin/opa
 
 install-opa:
-	sudo curl -s -L -o $(OPA_DEST) $(OPA_URL)
+	curl -s -L -O $(OPA_URL)
+	curl -s -L -O $(OPA_SHA_URL)
+	sha256sum -c $(OPA_SHA_FILE)
```
Comment (zregvart) [inline Makefile:36]: Something like this could work:
```suggestion
	echo "73ab60fab6c28131b4fa8f2e7e44973165c2be0c46e3e6feabce68d7eec9335a  opa_linux_amd64_static" | sha256sum --check
```

File: Makefile [hunk 3]
```diff
@@ -24,11 +24,17 @@ fmt-check:
 ci: fmt-check test
 
 OPA_VER=v0.39.0
-OPA_URL=https://openpolicyagent.org/downloads/$(OPA_VER)/opa_linux_amd64_static
+OPA_FILE=opa_linux_amd64_static
+OPA_URL=https://openpolicyagent.org/downloads/$(OPA_VER)/$(OPA_FILE)
+OPA_SHA_FILE=$(OPA_FILE).sha256
+OPA_SHA_URL=$(OPA_URL).sha256
 OPA_DEST=/usr/bin/opa
```
Comment (zregvart) [inline Makefile:31]: Perhaps like this so we don't need to `sudo`?
```suggestion
OPA_DEST=$HOME/opa
```

**Review by zregvart — COMMENTED** [review]
Some ideas...

File: Makefile (hunk 2 above)
Comment (simonbaird) [reply Makefile]: Thanks! Included in new revision.

**Review by zregvart — APPROVED** [review]
